lib/teams/aws-batch/aws-batch-on-eks-team.ts

```diff
@@ -0,0 +1,167 @@
+   * Allocation strategies for EKS Compute environment
+   */
+  allocationStrategy?: 'BEST_FIT_PROGRESSIVE' | 'SPOT_CAPACITY_OPTIMIZED'
+
+  jobQueueName: string
+
+  priority: number,
+}
+
+const defaultProps: BatchEksTeamProps = {
+  name: NAMESPACE,
+  namespace: NAMESPACE,
+  allocationStrategy: 'BEST_FIT_PROGRESSIVE',
+  jobQueueName: 'batch-eks-job',
+  priority: 10,
+}
+
+/*
+ *This class define the Team that can be used with AWS Batch on EKS
+ */
+
+export class BatchEksTeam extends ApplicationTeam {
+
+  private batchTeam: BatchEksTeamProps;
+  /**
+   * @public
+   * @param {BatchEksTeamProps} props the Batch team definition {@link BatchEksTeamProps}
+   */
+  constructor(props: BatchEksTeamProps) {
+    super({...defaultProps, ...props});
+    this.batchTeam = props;
+  }
+
+  setup(clusterInfo: ClusterInfo): void {
+    const cluster = clusterInfo.cluster;
+    const allocStr = this.batchTeam.allocationStrategy || 'BEST_FIT_PROGRESSIVE';
+
+    const AwsBatchAddOn = clusterInfo.getProvisionedAddOn('AwsBatchAddOn');
+
+    if (AwsBatchAddOn === undefined) {
+      throw new Error("AwsBatchAddOn must be deployed before creating AWS Batch on EKS team.");
+    }
+
+    // Set AWS Batch namespace and necessary RBACs
+    const ns = this.setBatchEksNamespace(clusterInfo, this.batchTeam.namespace);
+
+    // Create compute environment
+    const computeEnv = this.setComputeEnvironment(clusterInfo, this.batchTeam.namespace, ns, allocStr);
+
+    // Submit a job queue
+    const jobQueue = new batch.CfnJobQueue(clusterInfo.cluster.stack,'batch-eks-job-queue',{
+      jobQueueName: this.batchTeam.jobQueueName,
+      priority: this.batchTeam.priority,
+      computeEnvironmentOrder: [
+        {
+          order: 1,
+          computeEnvironment: computeEnv.attrComputeEnvironmentArn
+        }
+      ]
+    })
+
+    jobQueue.node.addDependency(computeEnv);
+  }
+  /**
+   * method to to apply k8s RBAC to the service account used by Batch service role
+   * @param ClusterInfo EKS cluster where to apply the RBAC
+   * @param namespace Namespace where the RBAC are applied
+   * @param createNamespace flag to create namespace if not already created
+   * @returns 
+   */
+
+  private setBatchEksNamespace(clusterInfo: ClusterInfo, namespace: string): eks.KubernetesManifest {
+
+    let doc = readYamlDocument(`${__dirname}/aws-batch-rbac-config.ytpl`);
+
+    //Get the RBAC definition and replace with the namespace provided by the user
+    const manifest = doc.split("---").map(e => loadYaml(e));
+
+    const values: Values = {
+      namespace: namespace
+    };
+
+    const manifestDeployment: ManifestDeployment = {
+      name: 'aws-batch-rbacs',
+      namespace: namespace,
+      manifest,
+      values
+    };
+
+    const kubectlProvider = new KubectlProvider(clusterInfo);
+    const statement = kubectlProvider.addManifest(manifestDeployment);
+
+    const batchEksNamespace = createNamespace(namespace, clusterInfo.cluster, true, true);
```

Comment:
why not put the namespace in the yptl file to start with, rather than deal with wiring dependency/order for execution here?

Reply:
addressed